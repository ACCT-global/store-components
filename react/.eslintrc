{
  "extends": "eslint-config-vtex-react",
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  }
}
