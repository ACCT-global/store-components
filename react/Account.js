export { default as default } from './components/Account'

