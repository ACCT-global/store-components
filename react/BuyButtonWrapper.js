import BuyButtonWrapper from './components/BuyButton/Wrapper'

export default BuyButtonWrapper