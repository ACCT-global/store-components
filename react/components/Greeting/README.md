# Greeting
Greeting is a canonical component that any VTEX app can import.

To import it into your code: 
```js
import Greeting from 'vtex.store-components/Greeting'
```

## Usage
You can use it in your code like a React component with the jsx tag: `<Greeting />`. 

Greeting will render a welcome message with the user first name if that is available on orderForm.
