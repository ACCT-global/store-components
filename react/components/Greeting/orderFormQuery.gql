query orderFromQuery {
  minicart @client {
    orderForm
  }
}
