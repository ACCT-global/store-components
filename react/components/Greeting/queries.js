import gql from 'graphql-tag'

export const orderFormQuery = gql`
  query {
    minicart @client {
      orderForm
    }
  }
`
