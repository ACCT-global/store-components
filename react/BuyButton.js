import { Component } from 'react'
import PropTypes from 'prop-types'
import { compose, graphql } from 'react-apollo'

import Spinner from '@vtex/styleguide/lib/Spinner'
import spinnerStyle from '@vtex/styleguide/lib/Spinner/style.css'

import productsQuery from './graphql/productsQuery.graphql'


const options = {
  options: ({
    category = '',
    specificationFilters = '',
    priceRange = '',
    collection = '',
    orderBy = '',
    from = 0,
    to = 8,
    salesChannel = '',
  }) => ({
    variables: {
      category,
      specificationFilters: specificationFilters ? [specificationFilters] : [],
      priceRange,
      collection,
      orderBy,
      from,
      to: to - 1,
      salesChannel,
    },
    ssr: false,
  }),
}


/**
 * BuyButton Component. Adds a list of itens to the cart.
 */
class BuyButton extends Component {

  constructor(props) {
    super(props)
    this.state = { data: { loading: true } }
  }
  
  componentDidMount() {
    this.setState( { data: graphql(productsQuery, options) } )
  }

  render() {
    const { data, maxItems, titleColor, titleText } = this.props
    const products = data['error'] ? [] : data.products
    const { sliderMounted } = this.state
    const slideSettings = this.configureSettings()

    return (
      <div className="ml7 mr7 pv4">
         { loading && (
          <div className="flex justify-around pa7">
            <div className="w3">
              <Spinner style={spinnerStyle} />
            </div>
          </div>
        )}
      </div>
    )
  }
}

Shelf.propTypes = {
  /** The quantity of products to be added to the cart */
  quantity: PropTypes.number.isRequired,
  /** The specification of wich product will be added to the cart */
  skuId:    PropTypes.string.isRequired,
  /** Wich seller is being referenced by the button */
  seller:   PropTypes.string.isRequired,
  /**  */
  salesChannel: PropTypes.string.isRequired,
  /** Should redirect or not */
  redirect: PropTypes.bool
}

export default BuyButton
