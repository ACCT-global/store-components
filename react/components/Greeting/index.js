import PropTypes from 'prop-types'
import React, { Fragment } from 'react'
import { compose, branch, renderComponent } from 'recompose'
import { FormattedMessage } from 'react-intl'
import { graphql } from 'react-apollo'
import { orderFormQuery } from './queries'

import Loader from './Loader'

import styles from './styles.css'

const Wrapper = ({ children }) => (
  <div
    className={`${
      styles.greetingContainer
    } mh4 pv4 t-heading-4 c-on-base nowrap`}
  >
    {children}
  </div>
)

const withWrapper = Component => props => (
  <Wrapper>
    <Component {...props} />
  </Wrapper>
)

const Greeting = ({ orderForm }) => {
  return (
    <Wrapper>
      <Fragment>
        <span className={styles.message}>
          <FormattedMessage id="greeting" />,
        </span>
        <span className={`${styles.orderForm} pl2 b`}>{orderForm}</span>
      </Fragment>
    </Wrapper>
  )
}

Greeting.propTypes = {
  orderForm: PropTypes.string,
}

const withLinkStateOrderForm = graphql(orderFormQuery, {
  props: ({ data: { minicart } }) => ({
    orderForm: minicart && minicart.orderForm && JSON.parse(minicart.orderForm),
  }),
})

export default compose(
  withLinkStateOrderForm,
  branch(({ loading }) => loading, renderComponent(withWrapper(Loader)))
)(Greeting)
