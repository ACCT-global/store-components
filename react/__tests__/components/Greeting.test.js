import { InMemoryCache } from 'apollo-cache-inmemory'
import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import { render } from '@vtex/test-tools/react'

import Greeting from '../../Greeting'

describe('<Greeting /> component', () => {
  const data = {
    minicart: {
      __typename: 'Minicart',
      orderForm: {
        __typename: 'OrderFormClient',
        clientProfileData: {
          __typename: 'ClientProfileData',
          firstName: 'name',
        },
      },
    },
  }
  const cache = new InMemoryCache()
  cache.writeData({ data })

  const renderComponent = customProps => {
    const props = {
      ...customProps,
    }

    const comp = (
      <MockedProvider cache={cache} resolvers={{}}>
        <Greeting {...props} />
      </MockedProvider>
    )

    return render(comp)
  }

  it('should be rendered', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot without loading', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with loading', () => {
    const { asFragment } = renderComponent({ loading: true })
    expect(asFragment()).toMatchSnapshot()
  })
})
