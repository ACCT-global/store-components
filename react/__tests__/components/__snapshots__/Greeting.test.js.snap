// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting /> component should match snapshot with loading 1`] = `
<div
  class="greetingContainer mh4 pv4 t-heading-4 c-on-base nowrap"
>
  <div
    class="content-loader-mock"
  >
    <rect
      height="32"
      rx="1"
      ry="1"
      width="300"
    />
  </div>
</div>
`;

exports[`<Greeting /> component should match snapshot without loading 1`] = `
<div
  class="greetingContainer mh4 pv4 t-heading-4 c-on-base nowrap"
>
  <span
    class="message"
  >
    <span>
      Hello
    </span>
    ,
  </span>
  <span
    class="firstName pl2 b"
  >
    name
  </span>
</div>
`;
